service/exporter/k8s/k8s.go

```diff
@@ -0,0 +1,386 @@
+package k8s
+
+import (
+	"k8s.io/api/apps/v1beta1"
+	"k8s.io/api/batch/v2alpha1"
+	"k8s.io/api/core/v1"
+	v1b1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/microerror"
+	micrologger "github.com/giantswarm/microkit/logger"
+)
+
+type metadata struct {
+	Labels map[string]string
+}
+
+type Service struct {
+	metadata
+	Name     string
+	Ports    []v1.ServicePort
+	Type     v1.ServiceType
+	Selector map[string]string
+}
+
+type Deployment struct {
+	metadata
+	Name   string
+	Status v1beta1.DeploymentStatus
+}
+
+type Pod struct {
+	metadata
+	Name              string
+	Status            string
+	ContainerStatuses []v1.ContainerStatus
+}
+
+type PodTemplate struct {
+	metadata
+	Containers    []v1.Container
+	RestartPolicy string
+	DNSPolicy     string
+}
+
+type DaemonSet struct {
+	metadata
+	Status      v1b1.DaemonSetStatus
+	PodTemplate PodTemplate
+	Selector    metav1.LabelSelector
+}
+
+type StatefulSet struct {
+	metadata
+	ServiceName string
+	Replicas    int32
+	PodTemplate PodTemplate
+	Selector    metav1.LabelSelector
+	Status      v1beta1.StatefulSetStatus
+}
+
+type CronJob struct {
+	metadata
+	Name        string
+	Schedule    string
+	Status      v2alpha1.CronJobStatus
+	Suspend     bool
+	JobTemplate JobTemplate
+}
+
+type JobTemplate struct {
+	metadata
+	PodTemplate PodTemplate
+}
+
+type Ingress struct {
+	metadata
+	Backends v1b1.IngressBackend
+	Rules    []v1b1.IngressRule
+	TLSHosts []v1b1.IngressTLS
+	Status   v1b1.IngressStatus
+}
+
+type Namespace struct {
+	metadata
+	Name            string
+	Pods            []Pod
+	Deployments     []Deployment
+	Services        []Service
+	DaemonSet       []DaemonSet
+	StatefulSets    []StatefulSet
+	CronJobs        []CronJob
+	Ingresses       []Ingress
+	NetworkPolicies []NetworkPolicy
+}
+
+func GetNamespaces(c *kubernetes.Clientset, excludes []string, log micrologger.Logger) []Namespace {
+	// creates the in-cluster config
+	ns, err := c.Namespaces().List(metav1.ListOptions{})
+	if err != nil {
+		//TODO use other than panic
+		panic(err.Error())
```

Comment:
This looks like proper error handling should be added. Same for the debug logs below. When there are errors we should handle them and forward them to the caller. When these are no critical errors their messages should maybe just logged as warnings.